### PR TITLE
Clarify how to pin the Sphinx version

### PR DESCRIPTION
**Pull request by astrojuanlu**
See https://github.com/readthedocs/readthedocs.org/issues/8679.

### CODE REVIEW

**Comment by astrojuanlu** [comment]
:robot: Rendered version: https://docs--8687.org.readthedocs.build/en/8687/config-file/v2.html#sphinx

File: docs/config-file/v2.rst
Comment (stsewd) [inline docs/config-file/v2.rst:444]:
```suggestion

   It is recommended to pin Sphinx to a specific version using a ``requirements.txt`` or ``environment.yml`` file
```

Comment (stsewd) [reply docs/config-file/v2.rst:444]: We should also mention this for MkDocs

Comment (astrojuanlu) [reply docs/config-file/v2.rst:444]: Good catch! Adding

File: docs/config-file/v2.rst
Comment (agjohnson) [inline docs/config-file/v2.rst:447]: I'd really like to have a doc dedicated to pinning dependencies that covers all of the nice patterns we support. This would be a good doc to go over nice packaging patterns like extra requires and using poetry.

Our config docs are the best we have right now though, but it is a bit obtuse from a user perspective.

Also most users just listen to our advice, so telling users to "use a requirements.txt file" is a bit hostile to package maintainers, which don't use this concept in packaging. We've had this push back from users before, where package maintainers want to use concepts they are already using in packaging, like extra requires.

Not something that needs to hold up this PR, but it would be nice to return to this later, change the language to be a bit more general, and point to a feature page dedicated to pinning dependencies. Pinning dependencies is a common enough part of our configuration, and users have enough problems understanding it, that we really should have top level docs on this.

**Review by agjohnson — APPROVED** [review]
Looks good! Is a pinning dependencies feature doc perhaps something we should put on our roadmap? Or perhaps expanding just the poetry docs we've been discussing?

File: docs/config-file/v2.rst
Comment (astrojuanlu) [reply docs/config-file/v2.rst:447]: I see your point about requirements files. As a library maintainer, I don't want to duplicate dependencies in a metadata file with abstract dependencies plus a list of concrete dependencies. (On abstract vs concrete: Donald Stufft explained everything here https://caremad.io/posts/2013/07/setup-vs-requirement/)

I will tweak the language to make it more amenable for other users, while making it clear that concrete dependencies are better suited for folks publishing docs that are not read from a reusable Python library.

**Comment by astrojuanlu** [comment]
> Is a pinning dependencies feature doc perhaps something we should put on our roadmap? Or perhaps expanding just the poetry docs we've been discussing?

You mean a "how packaging works in Python"? That's an interesting can of worms... 😃 

Now seriously, we can discuss having a guide on the topic. The current landscape of PEPs and tools seems confusing, but actually things are much easier these days than it used to be.

**Comment by astrojuanlu** [comment]
Enabling auto-merge on this one

**Pull request merged**
